tests/integration/devfile/cmd_devfile_create_test.go

```diff
@@ -191,9 +191,21 @@ var _ = Describe("odo devfile create command tests", func() {
 				helper.CmdShouldPass("odo", "create", "nodejs")
 				output := helper.CmdShouldFail("odo", "create", "nodejs")
 				Expect(output).To(ContainSubstring("this directory already contains a component"))
-				output = helper.CmdShouldFail("odo", "create", "nodejs", "--s2i")
-				Expect(output).To(ContainSubstring("this directory already contains a component"))
+			})
+		})
+
+		Context("Testing Create for OpenShift specific scenarios", func() {
+			JustBeforeEach(func() {
+				if os.Getenv("KUBERNETES") == "true" {
+					Skip("This is a OpenShift specific scenario, skipping")
+				}
+			})
 
+			It("should fail when we create the devfile or s2i component multiple times", func() {
+				helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", devfile), filepath.Join(commonVar.Context, devfile))
+				helper.CmdShouldPass("odo", "create", "nodejs")
+				output := helper.CmdShouldFail("odo", "create", "nodejs", "--s2i")
+				Expect(output).To(ContainSubstring("this directory already contains a component"))
```

Comment:
This tests say "should fail when we create the devfile or s2i component multiple times", however here only create devfile component once then try to create a s2i component.
To fully cover the possible paths we should:
1. Create devfile component (should pass) then try to create again a devfile component (should fail), try to create a s21 component (should fail)
2. Create a s2i component  (should pass) then try to create again a s21 component (should fail), try to create a devfile component (should fail)
With this then we can get rid of test in lines 190-193

Reply:
> With this then we can get rid of test in lines 190-193

Line number 190-193 is for both openshift and kubernetes scenario I separated out s2i from there because s2i images won't be available for kubernetes. 
Lines 204-208 is only openshift specific scenario. So in brief for openshift we are testing two scenario:

1. Create devfile component (should pass) then create devfile component again (should fail).
2. Create devfile component (should pass) then create s2i component (should fail). As we have already tested with twice devfile creation, checking it again will duplicate the things.